chain/chain/src/state_snapshot_actor.rs

```diff
@@ -176,26 +220,38 @@ pub struct SnapshotCallbacks {
 }
 
 /// Sends a request to make a state snapshot.
+// TODO: remove the `prev_block_hash` argument. It's just block.header().prev_hash()
 pub fn get_make_snapshot_callback(
     sender: StateSnapshotSenderForClient,
     flat_storage_manager: FlatStorageManager,
 ) -> MakeSnapshotCallback {
-    Arc::new(move |prev_block_hash, epoch_height, shard_indexes_and_uids, block| {
-        tracing::info!(
+    Arc::new(
+        move |prev_block_hash,
+              min_chunk_prev_height,
+              epoch_height,
+              shard_indexes_and_uids,
+              block| {
+            tracing::info!(
             target: "state_snapshot",
             ?prev_block_hash,
             ?shard_indexes_and_uids,
             "make_snapshot_callback sends `DeleteAndMaybeCreateSnapshotRequest` to state_snapshot_addr");
-        // We need to stop flat head updates synchronously in the client thread.
-        // Async update in state_snapshot_actor and potentially lead to flat head progressing beyond prev_block_hash
-        // This also prevents post-resharding flat storage catchup from advancing past `prev_block_hash`
-        flat_storage_manager.want_snapshot(prev_block_hash);
```

Comment:
also pretty sure this was a bug before. wasnt being triggered, but we want to stop at the min prev_block_hash for all chunks, not just the height of prev prev hash.